src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -76,6 +77,78 @@ protected virtual HttpClient GetHttpClient()
             return await ExecuteWithRetryAsync(uri, headers, body, HttpMethod.Post, logger, doNotThrow: true).ConfigureAwait(false);
         }
 
+        public async Task<HttpResponse> SendGetWithTimeoutAsync(
```

Comment:
Can you try to modify the existing methods to take in a CancellationToken instead ? There is too much duplicate code that you've written.